### ISSUE TITLE
[dr-autosync] pitr checkpoint ts lag up to 15min after down backup dc 

### ISSUE THREAD

**Issue by mayjiang0203** [tikv/tikv#15279]
### What version of TiKV are you using?

sh-4.2# /tiup/deploy/tikv-20160/bin/tikv-server -V
TiKV
Release Version:   6.5.3
Edition:           Community
Git Commit Hash:   852feae645973d2cca58c66fef99c32f95ccd1d3
Git Commit Branch: heads/refs/tags/v6.5.0_fix_flashback
UTC Build Time:    2023-07-27 07:59:32
Rust Version:      rustc 1.67.0-nightly (96ddd32c4 2022-11-14)
Enable Features:   pprof-fp jemalloc mem-profiling portable sse test-engine-kv-rocksdb test-engine-raft-raft-engine cloud-aws cloud-gcp cloud-azure
Profile:           dist_release

### Steps to reproduce

```
[2023/08/03 14:47:05.658 +08:00] [INFO] [cluster.go:516] ["network partition will be inject"] [source="[tiup,tidb1,pd1,pd2,tikv1,tikv2,tikv3,benchtoolset,br,minio]"] [target="[tikv4,tikv5,tikv6,tidb2,pd3]"]
[2023/08/03 14:47:05.786 +08:00] [INFO] [util.go:68] ["waiting for workload running after chaos injected"] [wait=10m0s] [callBy=primary.go:188]
[2023/08/03 14:58:06.047 +08:00] [INFO] [cluster.go:516] ["network partition will be inject"] [source="[tiup,tikv1,tikv2,tikv3,tidb1,pd1,pd2,benchtoolset,br,minio]"] [target="[pd3,tikv4,tikv5,tikv6,tidb2]"]
[2023/08/03 14:58:06.426 +08:00] [INFO] [util.go:68] ["waiting for workload running after chaos injected"] [wait=10m0s] [callBy=primary.go:188]
[2023/08/03 15:09:06.672 +08:00] [INFO] [cluster.go:516] ["network partition will be inject"] [source="[tiup,tikv1,tikv2,tikv3,tidb1,pd1,pd2,benchtoolset,br,minio]"] [target="[tikv4,tikv5,tikv6,tidb2,pd3]"]
[2023/08/03 15:09:06.803 +08:00] [INFO] [util.go:68] ["waiting for workload running after chaos injected"] [wait=10m0s] [callBy=primary.go:188]
[2023/08/03 15:20:07.051 +08:00] [INFO] [cluster.go:516] ["network partition will be inject"] [source="[tiup,pd1,pd2,tikv1,tikv2,tikv3,tidb1,benchtoolset,br,minio]"] [target="[tidb2,pd3,tikv4,tikv5,tikv6]"]
[2023/08/03 15:20:07.687 +08:00] [INFO] [util.go:68] ["waiting for workload running after chaos injected"] [wait=10m0s] [callBy=primary.go:188]
```

### What did you expect?
```
The gap should not grow continually after down backup dc.
```

### What did happened?
![image](https://github.com/tikv/tikv/assets/9443637/7bfad580-29fc-4d2a-a6da-cb28ac8ca6c9)

**Comment by BornChanger** [comment]
please add component label to this issue @mayjiang0203, I failed to add the tag

**Comment by YuJuncen** [comment]
During the test result, this problem doesn't really impact release-7.1. But we are still able to bring some optimization to 7.1.x versions. So the `affects-7.1` label has been added.

**Comment by mayjiang0203** [comment]
With the fix testing on 6.5, didn't hit this in the cluster with 16K regions on each store，
![0745c48c-7958-44c7-8f59-3f11ead0581b](https://github.com/tikv/tikv/assets/9443637/ab850e63-40a0-46ef-9fb1-f49a734d5816)
while still hit this in the cluster with 73 regions on each store,
![image](https://github.com/tikv/tikv/assets/9443637/f3e1dcdb-647b-4e05-99c0-0c68dc910b63)
